beep/protocol_tools/arbin_schedule_file.py

```diff
@@ -0,0 +1,251 @@
+        DOES NOT check the flow control or limits set in the steps. The dictionary
+        must represent a valid schedule before it is passed to this function.
+
+        Args:
+            dict_obj (dict): Ordered dictionary containing all of the schedule file
+                sections with keys and values. Nested dicts correspond to sections
+            outputfile (str): File string corresponding to the file to output the schedule to
+        """
+        f = open(outputfile, 'wb')
+        dict_obj.keys()
+        for key_line in dict_obj.keys():
+            if re.search(self.section, key_line) is not None:
+                line = key_line + '\r\n'
+                f.write(line.encode(self.schedule_file_encoding))
+                key_lines_1 = dict_obj[key_line].keys()
+                for key_line_1 in key_lines_1:
+
+                    if re.search(self.step, key_line_1) is not None:
+                        line = key_line_1 + '\r\n'
+                        f.write(line.encode(self.schedule_file_encoding))
+                        key_lines_2 = dict_obj[key_line][key_line_1].keys()
+                        for key_line_2 in key_lines_2:
+                            if re.search(self.limit, key_line_2) is not None:
+                                line = key_line_2 + '\r\n'
+                                f.write(line.encode(self.schedule_file_encoding))
+                                for key_line_3 in dict_obj[key_line][key_line_1][key_line_2].keys():
+                                    line = key_line_3 + '=' + \
+                                           dict_obj[key_line][key_line_1][key_line_2][key_line_3] + \
+                                           '\r\n'
+                                    f.write(line.encode(self.schedule_file_encoding))
+                            else:
+                                line = key_line_2 + '=' + \
+                                       dict_obj[key_line][key_line_1][key_line_2] + \
+                                       '\r\n'
+                                f.write(line.encode(self.schedule_file_encoding))
+                    else:
+                        line = key_line_1 + '=' + \
+                               dict_obj[key_line][key_line_1] + \
+                               '\r\n'
+                        f.write(line.encode(self.schedule_file_encoding))
+        f.close()
+
+    def fast_charge_file(self, CC1, CC1_capacity, CC2, inputname, outputname):
+        """
+        Function takes parameters for the FastCharge Project
+        and creates the schedule files necessary to run each of
+        these parameter combinations. Assumes that control type
+        is CCCV.
+
+        Args:
+            CC1 (float): Constant current value for charge section 1
+            CC1_capacity (float): Capacity to charge to for section 1
+            CC2 (float): Constant current value for charge section 2
```

Comment:
Might help to have units in the docstring: A/Ah?

Reply:
Yup! thanks for catching that